### PR TITLE
[hplsql] hplsql_session reuse solution

### PR DESCRIPTION
**Pull request by agl29**
## What changes were proposed in this pull request?

- Solving the session_reuse problem
- adding application='hplsql' for HPLSQL mode

### CODE REVIEW

**Comment by agl29** [comment]
needed to add the application='hplsql' otherwise, it will use the simple HIVE mode. (https://github.com/cloudera/hue/pull/2576/files#diff-95ac03d3f59e20ed88fa520fef4c4235e09158ff94327d7a859f6991baafdea8R181)

Any other good way to solve the problem of reuse_session? @romainr

File: apps/beeswax/src/beeswax/server/hive_server2_lib.py [hunk 1]
```diff
@@ -710,7 +710,7 @@ def open_session(self, user):
 
     session = Session.objects.create(
         owner=user,
-        application=self.query_server['server_name'],
+        application='hplsql' if HPLSQL.get() and self.query_server['server_name'] == 'beeswax' else self.query_server['server_name'],
```
Comment (romainr) [inline apps/beeswax/src/beeswax/server/hive_server2_lib.py:713]: Avoid one liner here and do above?

`application=....`

**Review by romainr — APPROVED** [review]
Feels kind of brittle, isn't there a way to just use `HPLSQL.get()` throughout?
(i.e. we don't need the 'hpsql' name)

**Comment by agl29** [comment]
What I found is that
1-> So [here](https://github.com/cloudera/hue/blob/08422587e8dc04501c954eabe349f96eb852acdc/desktop/libs/notebook/src/notebook/connectors/hiveserver2.py#L188) we are searching for session in Session model.
2 -> and [here](https://github.com/cloudera/hue/blob/master/apps/beeswax/src/beeswax/models.py#L401) need to put some different application name, then only it separates the sessions of hive and hplsql. otherwise how it will differentiate between them?
@romainr

**Pull request merged**